pkg/expression/builtin_time.go

```diff
@@ -2864,17 +2864,55 @@ func (du *baseDateArithmetical) getIntervalFromString(ctx sessionctx.Context, ar
 	if isNull || err != nil {
 		return "", true, err
 	}
-	// unit "DAY" and "HOUR" has to be specially handled.
-	if toLower := strings.ToLower(unit); toLower == "day" || toLower == "hour" {
-		if strings.ToLower(interval) == "true" {
-			interval = "1"
-		} else if strings.ToLower(interval) == "false" {
+
+	interval, err = du.intervalReformatString(ctx.GetSessionVars().StmtCtx.TypeCtx(), interval, unit)
+	return interval, false, err
+}
+
+func (du *baseDateArithmetical) intervalReformatString(tc types.Context, str string, unit string) (interval string, err error) {
```

Comment:
How about using `errctx`, rather than the `HandleTruncate` in `types.Context`?

Reply:
done